core/src/main/kotlin/org/evomaster/core/EMConfig.kt

```diff
@@ -550,4 +550,103 @@ class EMConfig {
             "NOTE: this should not cause any tests to fail.")
     var enableBasicAssertions = false
 
+    enum class ResourceSamplingStrategy (val requiredArchive : Boolean = false){
+        NONE,
+        /**
+         * probability for applicable strategy is specified
+         */
+        Customized,
+        /**
+         * probability for applicable strategy is equal
+         */
+        EqualProbability,
+        /**
+         * probability for applicable strategy is derived based on actions
+         */
+        Actions,
+        /**
+         * probability for applicable strategy is adaptive with time
+         */
+        TimeBudgets,
+        /**
+         * probability for applicable strategy is adaptive with performance, i.e., Archive
+         */
+        Archive (true),
+        /**
+         * probability for applicable strategy is adaptive with performance, i.e., Archive
+         */
+        ConArchive (true)
+    }
+
+    @Experimental
+    @Cfg("Specify whether to enable resource-based strategy to sample an individual during search. " +
+            "Note that resource-based sampling is only applicable for REST problem with MIO algorithm.")
+    var resourceSampleStrategy = ResourceSamplingStrategy.NONE
+
+    @Experimental
+    @Cfg("Specify whether to enable resource dependency heuristics, i.e, probOfEnablingResourceDependencyHeuristics > 0.0. " +
+            "Note that the option is available to be enabled only if resource-based smart sampling is enable. " +
+            "This option has an effect on sampling multiple resources and mutating a structure of an individual.")
+    var probOfEnablingResourceDependencyHeuristics = 0.5
+
+    @Experimental
+    @Cfg("Specify whether to export derived dependencies among resources. " +
+            "if enabled, the dependencies are stored at the same location of generated test cases.")
+    var exportDependencies = false
```

Comment:
this only applies to REST resources, isn't it? in that case, clarify it in the description